compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -595,20 +595,15 @@ object ExtractSemanticDB:
 
   def write(source: SourceFile, occurrences: List[SymbolOccurrence], symbolInfos: List[SymbolInformation])(using Context): Unit =
     def absolutePath(path: Path): Path = path.toAbsolutePath.normalize
-    def common(root: Path, i1: java.util.Iterator[Path], i2: java.util.Iterator[Path]) =
-      require(root != null)
-      var res: Path = root
-      var next: Path = null
-      while
-        i1.hasNext && i2.hasNext
-        && { next = i1.next; next } == i2.next
-      do res = res.resolve(next)
-      res
-    end common
+    def commonPrefix[T](z: T)(i1: Iterable[T], i2: Iterable[T])(app: (T, T) => T): T =
+      (i1 lazyZip i2).takeWhile(p => p(0) == p(1)).map(_(0)).foldLeft(z)(app)
     val sourcePath = absolutePath(source.file.jpath)
     val sourceRoot =
+      // Here if `sourceRoot` and `sourcePath` do not share a common prefix then `relPath` will not be normalised,
```

Comment:
What if we call `.normalize` on `relPath` instead?

Reply:
this didn't work because you'd get something like
for the source file `/var/tmp/foo/T/bar/Src1.scala` and `sourceRoot` `Users/jamie/workspace/dotty/`  you would get relPath
`../../../../var/tmp/foo/T/bar/Src1.scala` which isn't good when you append it to
`Users/jamie/workspace/dotty/out/META-INF/semanticdb`

Reply:
OK I see, but it sounds like we should be running these tests with `-sourceroot /var/tmp/foo/` or something then, otherwise the files won't be placed in the correct subdirectory inside META-INF/semanticdb ?

Reply:
in the situation above, sourceRoot becomes `/` so its safe for all cases, the storage path for the semanticdb file then becomes `Users/jamie/workspace/dotty/out/META-INF/semanticdb/var/tmp/foo/T/bar/Src1.scala.semanticdb`

Reply:
explicit sourcePath in this situation could be used to reduce the number of nested directories but isnt critical